src/univariate/continuous/pgeneralizedgaussian.jl

```diff
@@ -141,9 +141,20 @@ function rand(rng::AbstractRNG, d::PGeneralizedGaussian)
     inv_p = inv(d.p)
     g = Gamma(inv_p, 1)
     z = d.α * rand(rng, g)^inv_p
-    if rand(rng) < 0.5
+    if rand(rng, Bool)
         return d.μ - z
     else
         return d.μ + z
     end
 end
+
+function rand!(rng::AbstractRNG, d::PGeneralizedGaussian, A::AbstractArray{<:Real})
```

Comment:
This function suffers from the same problems as mentioned above, can you remove it from the PR?